### DISCUSSION TITLE
Choose between pushState and replaceState for the query string

### DISCUSSION THREAD

**Discussion by thijsvdanker** [livewire/livewire#5365] [3. Feature ideas]
I'd love to be able to choose between history.pushState and history.replaceState when using the [Query String](https://laravel-livewire.com/docs/2.x/query-string) for each of the items in `$queryString`.

Something like:
```php
$queryString = [
  'foo' => ['history' => 'push'],
  'bar' => ['history' => 'replace']
];
```

# Use-case
Using the query string is great to share urls with others to replicate the state of the application.
It is annoying when you navigate to a data-table, play with some filters and want to go back.
This is especially notable when using the application on a mobile device.

**Comment by N1ebieski** [comment]
I also needed to force replaceState, so I found a (hopefully only temporary) solution to the problem > https://gist.github.com/N1ebieski/22d222d3ac3e01e6b97d6ab374139d0d
